src/EFCore.PG.NodaTime/NodaTimeMethodCallTranslator.cs

```diff
@@ -45,11 +46,41 @@ public class NodaTimeMethodCallTranslator : IMethodCallTranslator
         [NotNull] static readonly MethodInfo GetCurrentInstant =
             typeof(SystemClock).GetRuntimeMethod(nameof(SystemClock.GetCurrentInstant), Type.EmptyTypes);
 
+        /// <summary>
+        /// The mapping of supported method translations.
+        /// </summary>
+        [NotNull] static readonly Dictionary<MethodInfo, string> MethodInfoDatePartMapping = new Dictionary<MethodInfo, string>
+        {
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromYears),        new[] { typeof(int) }),  "years" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromMonths),       new[] { typeof(int) }),  "months" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromWeeks),        new[] { typeof(int) }),  "weeks" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromDays),         new[] { typeof(int) }),  "days" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromHours),        new[] { typeof(long) }), "hours" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromMinutes),      new[] { typeof(long) }), "mins" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromSeconds),      new[] { typeof(long) }), "secs" },
+            //{ typeof(Period).GetRuntimeMethod(nameof(Period.FromMilliseconds), new[] { typeof(long) }), "" },
+            //{ typeof(Period).GetRuntimeMethod(nameof(Period.FromNanoseconds),  new[] { typeof(long) }), "" },
+        };
+
         /// <inheritdoc />
         [CanBeNull]
-        public Expression Translate(MethodCallExpression expression)
-            => expression.Method == GetCurrentInstant
-                ? new AtTimeZoneExpression(new SqlFunctionExpression("NOW", expression.Type), "UTC", expression.Type)
-                : null;
+        public Expression Translate(MethodCallExpression e)
+        {
+            if (e.Method == GetCurrentInstant)
+                return new AtTimeZoneExpression(new SqlFunctionExpression("NOW", e.Type), "UTC", e.Type);
+
+            // TODO: Version compat? See DateTime.Add* translator
+            var declaringType = e.Method.DeclaringType;
+            if (declaringType == typeof(Period))
+            {
+                return MethodInfoDatePartMapping.TryGetValue(e.Method, out var datePart)
+                    ? new PgFunctionExpression("MAKE_INTERVAL", typeof(Period), new Dictionary<string, Expression> {
```

Comment:
> I would re-use declaringType here instead of typeof(Period). It doesn't make any difference right now, but if this gets refactored later on, I think tying the type to the check is more clear.

I understand, but when we add translations for other types they'll have their own separate code fragments - `Period.FromXX()` gets translated to `MAKE_INTERVAL()`, but other types and methods won't... So it makes sense to me to leave it hardcoded to `Period`.

> nit: In-line dictionaries are verbose... Is anything blocking us from adding a params (string, Expression)[] constructor to PgFunctionExpression?

Not really, but the calling syntax seems a bit weird to me:

```c#
new PgFunctionExpression("MAKE_INTERVAL", typeof(Period), (datePart, e.Arguments[0]), (otherParam, otherValue));
```

It seems more unclear/confusing then the dictionary syntax... In any case I don't think we have enough usage of `PgFunctionExpression` at the moment to care...